src/box/lua/experimental/connpool.lua

```diff
@@ -54,6 +54,45 @@ local function connect_using_config(instance_name, opts)
     return conn
 end
 
+local function is_labels_match(labels, instance_labels)
+    if labels == nil then
+        return true
+    end
+    if instance_labels == nil then
+        return false
+    end
+    for _, label in pairs(labels) do
+        if instance_labels[label] == nil then
+            return false
+        end
+    end
+    return true
+end
+
+local function get_candidates(role_name, opts)
```

Comment:
In a voice discussion with Mergen we agreed that filtering by a role should be one of the filters, not some kind of a primary filter in a separate argument.

So, we likely will be somethings like the following.

```lua
pool.get_candidates({
    role = 'myrole',
    labels = {zone = 'msk'},
})
```

Or the following.

```lua
pool.get_candidates({
    roles = {'myrole1', 'myrole2'},
    labels = {zone = 'msk'},
})
```